File: projects/0757ea4c-5bb8-4630-82c5-5d15f380c447/archive/20230816_162821/workspace/oauth2_client.py
from api_client import APIClient

class OAuth2Client(APIClient):
    def __init__(self, base_url, token_url, client_id, client_secret, scope):
        super().__init__(base_url)
        self.token_url = token_url
        self.client_id = client_id
        self.client_secret = client_secret
        self.scope = scope
        self.token = None

    def authenticate(self):
        data = {
            'client_id': self.client_id,
            'client_secret': self.client_secret,
            'scope': self.scope,
            'grant_type': 'client_credentials'
        }
        response = self.post(self.token_url, data=data)
        response.raise_for_status()
        self.token = response.json()['access_token']

    def get(self, path, headers=None, params=None):
        if self.token is None:
            self.authenticate()
        if headers is None:
            headers = {}
        headers['Authorization'] = f'Bearer {self.token}'
        return super().get(path, headers=headers, params=params)

    def post(self, path, headers=None, data=None):
        if self.token is None:
            self.authenticate()
        if headers is None:
            headers = {}
        headers['Authorization'] = f'Bearer {self.token}'
        return super().post(path, headers=headers, data=data)

File: projects/9ee52fd7-13e2-4fa0-9830-79d5543e1e32/archive/20230815_134247/workspace/OAuth2Client.js
const axios = require('axios');

class OAuth2Client {
  constructor(clientId, clientSecret, tokenUrl) {
    this.clientId = clientId;
    this.clientSecret = clientSecret;
    this.tokenUrl = tokenUrl;
    this.token = null;
  }

  async getAccessToken() {
    if (!this.token) {
      const response = await axios.post(this.tokenUrl, {
        client_id: this.clientId,
        client_secret: this.clientSecret,
        grant_type: 'client_credentials'
      });

      this.token = response.data.access_token;
    }

    return this.token;
  }
}

module.exports = OAuth2Client;
